### DISCUSSION TITLE
Encrypt images

### DISCUSSION THREAD

**Discussion by alangrainger** [alangrainger/share-note#124] [Vote for upcoming features]
Suggested by @pedramamini

> While text is encrypted, images are not. Optimally, they would be. At the moment we benefit from security through obscurity since the image paths are not known. As a stop gap, could improve randomness there

Comment from me:

Currently the image/asset filenames are derived from the hash of the content. This is very useful, as it means that multiple notes sharing the same content don't need to upload multiple copies of it.

Since the filenames are a 20 random character hash, the security-via-obscurity is very high. And if someone is able to access the content of your note to find out the filenames ... well then any encryption is moot.

Discussion is welcome.

**Comment by aiproductguy** [comment]
Beautiful! Content-addressible storage has so many advantages. Do you SHA-256 hash everything like so?

I am not sure if encrypting all images is the product solution. I would much rather see support for user-prompted/password encryption like Meld Encrypt (https://github.com/meld-cp/obsidian-encrypt).

**Comment by TangentFoxy** [comment]
For anyone concerned about leaking an image but insisting they must include it, [base64 encode your image and put it in an `img` tag](https://www.w3docs.com/snippets/html/how-to-display-base64-images-in-html.html).
